openfisca_uk/variables/finance/benefit/family/pension_credit.py

```diff
@@ -10,31 +10,38 @@ class pension_credit_reported(Variable):
     definition_period = YEAR
 
 
+class would_claim_PC(Variable):
+    value_type = bool
+    entity = BenUnit
+    label = u"Would claim Pension Credit if eligible"
+    definition_period = YEAR
+
+    def formula(benunit, period, parameters):
+        takeup_rate = parameters(period).benefit.pension_credit.takeup
+        return random(benunit) < takeup_rate
+
+
 class claims_PC(Variable):
     value_type = bool
     entity = BenUnit
     label = u"Whether this family is imputed to claim Pension Credit"
     definition_period = YEAR
 
     def formula(benunit, period, parameters):
-        already_claiming = (
-            aggr(benunit, period, ["pension_credit_reported"]) > 0
-        )
-        takeup_rate = parameters(period).benefit.pension_credit.takeup
-        return already_claiming + (random(benunit) < takeup_rate)
+        return benunit("would_claim_PC", period)
```

Comment:
Does this need to be separate from `would_claim_PC` because one is overridden by the microdata?